Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Commands/Commands.cs

```diff
@@ -130,66 +131,140 @@ public static Command CutActionsOrBindings()
             return (in InputActionsEditorState state) =>
             {
                 CopyPasteHelper.Cut(state);
-                return state.selectionType == SelectionType.Action ?
-                    DeleteAction(state.selectedActionMapIndex, Selectors.GetSelectedAction(state)?.wrappedProperty.FindPropertyRelative(nameof(InputAction.m_Name)).stringValue).Invoke(state)
-                    : DeleteBinding(state.selectedActionMapIndex, state.selectedBindingIndex).Invoke(state);
+                return state.CutActionOrBinding();
             };
         }
 
-        public static Command PasteActionMaps()
+        public static Command PasteActionMaps(IReadOnlyList<IPasteListener> pasteListeners)
         {
             return (in InputActionsEditorState state) =>
             {
-                var lastPastedElement = CopyPasteHelper.PasteActionMapsFromClipboard(state);
+                var newIndex = -99;
+                if (state.GetCutElements() != null && state.GetCutElements().Any())
+                    newIndex = CopyPasteHelper.DeleteCutElements(state);
+                else
+                {
+                    foreach (var pasteListener in pasteListeners)
+                        pasteListener.OnPaste(state);
+                }
+                var lastPastedElement = CopyPasteHelper.PasteActionMapsFromClipboard(state.With(selectedActionMapIndex: newIndex >= -1 ? newIndex : state.selectedActionMapIndex));
                 if (lastPastedElement != null)
                 {
                     state.serializedObject.ApplyModifiedProperties();
-                    return state.SelectActionMap(lastPastedElement.GetIndexOfArrayElement());
+                    return state.With(selectedActionMapIndex: lastPastedElement.GetIndexOfArrayElement(), cutElements: new List<CutElement>());
                 }
-                return state;
+                return state.With(cutElements: new List<CutElement>());
+            };
+        }
+
+        public static Command DeleteCutElements()
+        {
+            return (in InputActionsEditorState state) =>
+            {
+                CopyPasteHelper.DeleteCutElements(state);
```

Comment:
Should this just abort and return existing state if there are no elements to cut?

Reply:
added in [96c3270](https://github.com/Unity-Technologies/InputSystem/pull/1858/commits/96c32700fc4229f946838faf1eb88cdbacf631dc)